ThirdParty/bsp/freertos10_xilinx/src/Source/portable/GCC/ARM_CR5/port.c

```diff
@@ -156,6 +156,20 @@ debugger. */
  * a 32 bit status register
  */
 #define portFPU_REGISTER_WORDS ( (16 * 2) + 1)
+
+
+/* Adding the necessary stuff in order to be able to determine from C code wheter or not the IRQs are enabled at the processor level (not interrupt controller level) */
+#define GET_CPSR()	({u32 rval = 0U; \
+			  __asm__ __volatile__(\
+			    "mrs	%0, cpsr\n"\
+			    : "=r" (rval)\
+			  );\
+			  rval;\
+			 })
+
+#define CPSR_IRQ_ENABLE_MASK 0x80U
+
+#define IS_IRQ_DISABLED() ({unsigned int val = 0; val = (GET_CPSR() & CPSR_IRQ_ENABLE_MASK) ? 1 : 0; val;})
```

Comment:
Becauase in the FreeRTOS-Kernel repo the file pseudo_asm_gcc.h doesn't exist, there is no ready to use macro like mrcprs() to retreive the CPSR, thats why I redefine it here under an other nade